adminapi/app/static/react/admin.js

```diff
@@ -0,0 +1,412 @@
+                />
+                <PostList
+                    data={this.state.postList}
+                    handleSelect={this.handleSelect}
+                    getList={this.retrieveData}
+                />
+            </div>
+        );
+    }
+});
+
+var PostForm = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    getInitialState: function() {
+        return {
+            value: '',
+            fieldData: []
+        };
+    },
+    retrieveFieldData: function(data) {
+        var model = data['model'];
+        var app = data['app'];
+
+        $.ajax({
+            url: this.props.url + app + '/' + model + '/',
+            dataType: 'json',
+            type: 'GET',
+            beforeSend: function (xhr) {
+                xhr.setRequestHeader('Authorization', 'Token ' + Cookies.getJSON('token').token);
+            }.bind(this),
+            cache: false,
+            success: function(data) {
+                console.log(data);
+                this.setState({fieldData: data});
+            }.bind(this),
+            error: function(xhr, status, err) {
+                console.log(xhr.responseText);
+            }.bind(this)
+        });
+    },
+    handleSubmit: function(e) {
+        e.preventDefault();
+        var form = ReactDOM.findDOMNode(this.refs.posts);
+        var formData = new FormData(form);
+        if(!this.props.data.id) {
+        console.log('Create');
+        this.props.handleSubmit(
+            {model: 'post', app: 'post', formData: formData}
+        );
+        } else {
+        console.log('Update');
+        this.props.handleUpdate(
+            {id: this.props.data.id, model: 'post', app: 'post', formData: formData}
+        );
+        }
+    },
+    handleDelete: function(e) {
+        e.preventDefault();
+        console.log('Delete');
+        this.props.handleDelete(
+            {model: 'post', app: 'post', id: this.props.data.id}
+        );
+    },
+    handleCancel: function(e) {
+        e.preventDefault();
+        this.props.handleCancel({model: 'post'});
+    },
+    componentWillReceiveProps: function(nextProps) {
+        console.log('Props incoming');
+        var data ={};
+        for(var field in nextProps.data) {
+            data[field] = nextProps.data[field];
+        }
+        this.setState(
+           data
+        );
+        if(jQuery.isEmptyObject(nextProps.data)) {
+            var clearData = {};
+            for(var field in this.state) {
+                if(field != 'fieldData'){
+                    clearData[field] = '';
+                }
+            }
+            this.setState(clearData);
+        }
+    },
+    componentDidMount: function() {
+        this.retrieveFieldData({app: 'sites', model: 'site'});
+    },
+    render: function() {
+        var siteList = this.state.fieldData.map(function (site) {
+            return (
+                <option key={site.id} value={site.id}>{site.name}</option>
```

Comment:
React prefers if a list of items have key values, it will through a warning if they are not added. 

The single {} is used when providing a React HTML element with js data.